src/modules/components/data-visualisation/tabs-pane/tabs-pane.component.ts

```diff
@@ -17,18 +17,26 @@ export class TabsPaneComponent {
   selectedData: any;
   onClick: any;
 
+  static $inject = ['$timeout'];
+
   private selectedIndex: number;
 
-  constructor() {
-    if (this.defaultIndex != null) {
-      this.selectElement(this.defaultIndex);
-    }
+  constructor(private $timeout: ng.ITimeoutService) {
   }
 
   selectElement(elementIndex: number) {
     this.selectedIndex = elementIndex;
-    this.selectedData = this.elements[this.selectedIndex].data;
+    this.selectedData = this.elements.length > 0 ? this.elements[this.selectedIndex].data : [];
     this.onClick();
   }
 
+  $onChanges(changesObj: any) {
+    if (changesObj.defaultIndex && changesObj.defaultIndex.currentValue) {
```

Comment:
this would fail in case of null or 0... which might be valid values. Please check !== undefined